### PR TITLE
Remove filter functions

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Related to #4528 
Fixes `filter()` part of #4592 

### WHAT is this pull request doing?
Removing all instances of `filter()` Sass function because the color system used by this function has been deprecated. Swapping out any references to `filter()` with the [color system](https://github.com/Shopify/polaris-react/blob/main/documentation/Color%20system.md) custom css properties

## 

### How to 🎩
Most changes are in the `Navigation` component. Run Storybook and/or visit the [Chromatic](https://www.chromatic.com/build?appId=5d559397bae39100201eedc1&number=8945) link to this branch to ensure there are no visual regressions.

### CODE REVIEW

File: src/components/Navigation/Navigation.scss [hunk 1]
```diff
@@ -208,14 +208,14 @@ $disabled-fade: 0.6;
     @include recolor-icon(
```
Comment (alex-page) [inline src/components/Navigation/Navigation.scss:208]: Is this function used a bunch? I would assume it's a filter related function. If it is we could move it to this file.

**Review by alex-page — COMMENTED** [review]
I think we should remove the `filter()` function and point this to `v8.0.0-major` branch. Looking great so far! We will have to add documentation for migration and a breaking change to the changelog.

**Review by alex-page — APPROVED** [review]
Looks great for the first cut of removing the filter function references!

The files changed need a rebase/merge or something 👍

File: src/components/Navigation/Navigation.scss [hunk 2]
```diff
@@ -208,14 +208,14 @@ $disabled-fade: 0.6;
...
-      $filter-color: filter('action')
+      $filter-color: var(--p-action-primary)
     );
     background: var(--p-background-hovered);
 
     @media (-ms-high-contrast: active) {
       @include recolor-icon(
         $fill-color: var(--p-icon-on-interactive),
-        $filter-color: filter('white')
+        $filter-color: var(--p-icon-on-interactive)
```
Comment (BPScott) [inline src/components/Navigation/Navigation.scss:218]: These filter-color values passed to the 3rd argument of recolor-icon  are sent as the [`filter`](https://developer.mozilla.org/en-US/docs/Web/CSS/filter) property. They're not actual color values, they're filters like [`brightness(0) saturate(100%) invert(100%)`](https://github.com/Shopify/polaris-tokens/blob/main/dist/color-filters.color-map.scss#L138).

I think changing these to colors is going to break the behaviour, which is to allow for the recoloring of icons that are passed as svg datauris and rendered with an img tag

**Pull request merged**